### PR TITLE
Copy fake_joint_trajectory_controller from baxter_moveit_config

### PR DESCRIPTION
**Pull request by wkentaro**
As proposed in
    https://github.com/ros-planning/moveit_robots/pull/44#issuecomment

### CODE REVIEW

File: fake_joint_trajectory_controller/package.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  <maintainer email="mferguson@fetchrobotics.com">Michael Ferguson</maintainer>
+
+  <license>BSD</license>
+
+  <url type="website">http://moveit.ros.org</url>
+  <url type="bugtracker">https://github.com/ros-planning/moveit_ros/issues</url>
+  <url type="repository">https://github.com/ros-planning/moveit_ros</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
+
```
Comment (v4hn) [inline fake_joint_trajectory_controller/package.xml:20]: The script `run_depend`s on rospy, actionlib, actionlib_msgs, sensor_msgs, control_msgs and trajectory_msgs,
could you add these?

File: fake_joint_trajectory_controller/package.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  <export>
+  </export>
```
Comment (v4hn) [inline fake_joint_trajectory_controller/package.xml]: No need for these tags, is there?

File: fake_joint_trajectory_controller/package.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<package>
+  <name>fake_joint_trajectory_controller</name>
+  <version>0.7.2</version>
+  <description>Fake joint trajectory controller to run MoveIt without robot.</description>
+  <author email="isucan@google.com">Ioan Sucan</author>
+  <author email="robot.moveit@gmail.com">Sachin Chitta</author>
```
Comment (v4hn) [inline fake_joint_trajectory_controller/package.xml]: The copyright in the script says it's written by Kei Okada.
Do we have an email-address of him and shouldn't we list him as author?
If not, then maybe you should add your name there too?
At least it's not written by Ioan and Sachin.

File: fake_joint_trajectory_controller/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(fake_joint_trajectory_controller)
+find_package(catkin REQUIRED)
+catkin_package()
```
Comment (v4hn) [inline fake_joint_trajectory_controller/CMakeLists.txt:4]: The same applies as in ros-planning/moveit_robots#44.
Could you please add a `catkin_install_python` install rule for the script?

**Comment by v4hn** [comment]
Great, thanks for the request!
Could you please address the points I commented on?

**Comment by wkentaro** [comment]
Fixed.

**Comment by rhaschke** [comment]
I'm wondering why this controller doesn't filter the param `/move_group/controller_list` for its type.
As far as I can see, this makes it impossible to use this controller together with other ones in a multi-group system. However, maybe I missed something.

**Comment by davetcoleman** [comment]
This is certainly a needed feature. I'm concerned this is duplicate functionality as the FakeController and we should just have one or the other within the MoveIt! repos. I'm a fan of https://github.com/ros-planning/moveit_plugins/pull/20 because it simply enhances a previously existing and commonly used controller rather than add another one.

Another feature I would like to see is the ability to specify a robot pose from the SRDF as the initial starting state of the robot. I've found this a much needed feature in the past during my development.

I also want to point out a similar package I made a couple years ago - [moveit_sim_controller](https://github.com/davetcoleman/moveit_sim_controller) that does the same thing as both of these PRs. It already has the initial start state capability and even better - it is built on ros_control so you can use the same controllers and interfaces as you would on a real robot. However, it would add most of ros_control as a dependency to MoveIt! and I'm not interested in tackling making this the default fake controller for MoveIt! right now.

**Comment by v4hn** [comment]
Closing. This functionality is now provided in https://github.com/ros-planning/moveit_plugins/pull/23 in a more general and reusable way.

**Pull request closed without merging**